pkg/plugins/policies/core/matchers/dataplane.go

```diff
@@ -185,6 +188,20 @@ func ddpSelectedByNamespace(meta core_model.ResourceMeta, dpp *core_mesh.Datapla
 	}
 }
 
+func dppSelectedByZone(meta core_model.ResourceMeta, dpp *core_mesh.DataplaneResource) bool {
+	switch meta.GetLabels()[mesh_proto.PolicyRoleLabel] {
+	case string(mesh_proto.ConsumerPolicyRole), string(mesh_proto.WorkloadOwnerPolicyRole):
```

Comment:
`system` policies that originate on zone should also be prefiltered by the `zone`. Probably it's easier to check for `producer` here and do the filter for all other types:
```go
switch core_model.PolicyRole(meta) {
case mesh_proto.ProducerPolicyRole:
    return true
default:
    // check the zone label
    // ...
}
```

I think it's better to use `core_model.PolicyRole` because policies created before the upgrade don't have `kuma.io/policy-role` label.